tests/prefetch_related/tests.py

```diff
@@ -978,6 +978,51 @@ def test_nested_prefetch_related_with_duplicate_prefetcher(self):
         with self.assertNumQueries(5):
             self.traverse_qs(list(houses), [["occupants", "houses", "main_room"]])
 
+    def test_nested_prefetch_related_with_duplicate_prefetch_and_depth(self):
+        """
+        In the case of a defined to_attr, the order should not have an impact.
+
+        Ticket: #34791
+        """
+        people = Person.objects.all().prefetch_related(
+            Prefetch(
+                "houses__main_room",
+                queryset=Room.objects.filter(name="Dining room"),
+                to_attr="dining_room",
+            ),
+        )
+        with self.assertNumQueries(3):
+            main_room = people[0].houses.all()[0]
+
+        self.assertIs(hasattr(main_room, "dining_room"), True)
```

Comment:
Not convinced there's a lot of value in this regression assertion as it's already covered by plenty of tests.